### DISCUSSION TITLE
Dynamic libraries suport (single .hs file)

### DISCUSSION THREAD

**Discussion by luishgh** [haskell/haskell-language-server#1247] [Q&A] [answered]
I wanted to use haskell-language-server-wrapper inside CoC-vim to edit my xmonad configs. It is a single hs file and, because I use arch, all the haskell libraries in my computer are dynamic (including the Prelude). Because of this (at least I think that's the reason), I always get errors saying the libraries are missing. Is there any way to fix this? Thanks.

**Comment by jneira** [comment]
In the issues about dynamic linking the usual suggested workaround is built haskell-language-server itself with dynamic linking: https://github.com/haskell/haskell-language-server/issues/1160#issuecomment-756566273

**Comment by berberman** [comment, marked as answer]
Unfortunately, dynamic linking haskell-language-server can't solve this issue, because hls will use `defaultCradle` given by hie-bios to load xmonad.hs if there is no explicit cradle config (at least on my Arch Linux machine). However, `defaultCradle` contains an empty `componentOptions`, so hls can not know this standalone hs module should be loaded dynamically. Here is a related issue: https://github.com/mpickering/hie-bios/issues/257.

@felixonmars and I are working on this recently, and we decided to patch hie-bios. I believe hls can be installed directly from pacman on Arch Linux in the near future :)
